sentry/interceptor.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import asdict, is_dataclass
+from typing import Any, Optional, Type
+
+import sentry_sdk
+from temporalio import workflow
+from temporalio.worker import (
+    ActivityInboundInterceptor,
+    ExecuteActivityInput,
+    ExecuteWorkflowInput,
+    Interceptor,
+    WorkflowInboundInterceptor,
+    WorkflowInterceptorClassInput,
+)
+
+
+class _SentryActivityInboundInterceptor(ActivityInboundInterceptor):
+    async def execute_activity(self, input: ExecuteActivityInput) -> Any:
+        transaction_name = input.fn.__module__ + "." + input.fn.__name__
+        scope_ctx_manager = sentry_sdk.configure_scope()
+        with scope_ctx_manager as scope, sentry_sdk.start_transaction(
+            name=transaction_name
+        ):
+            sentry_sdk.set_tag("temporal.execution_type", "activity")
+            sentry_sdk.set_tag("temporal.activity.name", input.fn.__name__)
```

Comment:
```suggestion
            sentry_sdk.set_tag("temporal.activity.name", temporalio.activity.info().activity_type)
```

There are also other tags from info you might want like `activity_id` (so it can be tied w/ the Temporal history/UI), `workflow_type`, `workflow_id`, `workflow_namespace`, `task_queue`, etc.